io_scene_godot/converters/animation.py

```diff
@@ -0,0 +1,271 @@
+    time_offset = (frame_start - 1) * frame_len
+    # Godot only support linear interpolation, while Blender
+    # default use Bezier interpolation, so treat every frame
+    # as keyframe to increase precision
+    frame_start = int(frame_start)
+    frame_end = int(frame_end)
+    for frame in range(frame_start, frame_end+1):
+        bpy.context.scene.frame_set(frame)
+
+        timestamp = (frame - frame_start) * frame_len - time_offset
+        for path, obj in anim_objects.items():
+            track = tracks['.'.join([path, 'transform'])]
+
+            transform_mat = copy.deepcopy(fix_matrix(obj.matrix_local))
+            keyframe = KeyFrame(timestamp, transform_mat)
+
+            if frame == frame_start or frame == frame_end:
+                track.force_append_keyframe(keyframe)
+            else:
+                track.append_keyframe(keyframe)
+
+            if obj.type.lower() == 'armature':
+                if obj.pose is not None:
+                    for pbone in obj.pose.bones:
+                        track = tracks[
+                            '.'.join([path, pbone.name, 'transform'])
+                        ]
+
+                        pose_mat = copy.deepcopy(
+                            fix_matrix(pbone.matrix_basis)
+                        )
+                        keyframe = KeyFrame(timestamp, pose_mat)
+
+                        if frame == frame_start or frame == frame_end:
+                            track.force_append_keyframe(keyframe)
+                        else:
+                            track.append_keyframe(keyframe)
+
+    attach_tracks_to_anim_data(
+        animation_resource,
+        tracks.values(),
+        frame_len * (frame_end - frame_start + 1)
+    )
+
+    # FIXME
+    escn_file.internal_resources.append(animation_resource)
+    index = len(escn_file.internal_resources)
+    animation_resource.heading['id'] = index
+    return index
+
+
+def export_animation_player(escn_file, export_settings, parent_gd_node,
+                            anim_root_node, anim_objects):
+    """Generate an AnimationPlayer node from a map of animation objects"""
+    anim_player_node = NodeTemplate("AnimPlayer",
+                                    "AnimationPlayer", parent_gd_node)
+    anim_root_path = anim_root_node.get_path()
+    anim_player_node['root_node'] = NodePath(anim_root_path,
+                                             anim_player_node.get_path())
+
+    # TODO: shape key, constrain, nla_stack
+
+    orig_nla_status = []
+    orig_frame = bpy.context.scene.frame_current
+
+    for path, obj in anim_objects.items():
+        orig_nla_status.append(obj.animation_data.use_nla)
+        obj.animation_data.use_nla = False
+
+    anim_data_id = export_animation_data(escn_file, 'default',
+                                         anim_objects, anim_root_path)
+    anim_player_node['anims/default'] = "SubResource({})".format(anim_data_id)
+
+    # save back status
```

Comment:
I think blender stores and undoes any changes that the export script performs, so you don't need to worry about making changes to the blend.

Reply:
ok. I will remove it.